test/unit/rules/lint-no-childless-elements-test.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+
+generateRuleTests({
+  name: 'no-childless-elements',
+
+  config: true,
+
+  good: [
```

Comment:
The good and bad test cases can be generated by exporting `BLOCK_TAGS` from the rule file and then doing: 
```
good: BLOCK_TAGS.map(blockTag => {
  return ...
}),

bad: BLOCK_TAGS.map(blockTag => {
  return { ... };
})
```